aten/src/ATen/core/PhiloxRNGEngine.h

```diff
@@ -192,12 +190,55 @@ class philox_engine {
     ret[3] = lo0;
     return ret;
   }
+
+  C10_HOST_DEVICE constexpr float normalize(uint32_t value) {
+    int32_t x = (int32_t)value;
+    float scale = 4.6566127342e-10;
+    if (x < 0) {
+      x = -x - 1;
```

Comment:
On CPU a conditional like this are expensive, `& 0x7fffffff` should be faster.  I actually updated the version in torchdynamo to do this:
https://github.com/pytorch/torchdynamo/blob/748f99f75e8570a621a030d4a1eadcd29415c747/torchinductor/codegen/cpp.py#L81

Reply:
Neat :) will change